Source/Timeline/EventDescriptor.swift

```diff
@@ -5,13 +5,13 @@ import UIKit
 public protocol EventDescriptor: AnyObject {
   var startDate: Date {get set}
   var endDate: Date {get set}
-  var isAllDay: Bool {get}
-  var text: String {get}
+  var isAllDay: Bool {get set} // TODO: should we return this to read-only?
```

Comment:
Yes, these properties should be read-only, as the CalendarKit doesn't modify them.

Reply:
But in this case, we cannot use `EventDescriptor` as a generic type for custom descriptors. I find it quite convenient:
```swift
var event: EventDescriptor
if (Bool.random()) {
    event = Event()
} else {
    event = MyEvent()
}
// umposible with read-only properties
event.text = info.reduce("", {$0 + $1 + "\n"})
event.color = colors[Int(arc4random_uniform(UInt32(colors.count)))]
```

Reply:
Yes, but that's not a requirement by the library.
You're free to create own, stricter protocol requirement and use it. Meanwhile, it would inherit from `EventDescriptor`, so it would be perfectly compatible.